sdk/notificationhubs/notification-hubs/src/api/clientContext.ts

```diff
@@ -69,46 +54,44 @@ export function createClientContext(
   return new NotificationHubsServiceClient(connectionString, hubName, options);
 }
 
-class NotificationHubsServiceClient extends ServiceClient implements NotificationHubsClientContext {
-  sasTokenProvider: SasTokenProvider;
+class NotificationHubsServiceClient implements NotificationHubsClientContext {
+  sasTokenCredential: SasTokenCredential;
   baseUrl: string;
   hubName: string;
+  client: Client;
+  httpClient: HttpClient;
 
   constructor(
     connectionString: string,
     hubName: string,
     options: NotificationHubsClientOptions = {}
   ) {
-    super({
-      deserializationOptions: {
-        parseXML,
-      },
-      serializationOptions: {
-        stringifyXML,
-      },
-      userAgentOptions: {
-        userAgentPrefix: `azsdk-js-messaging-notificationhubs/${constants.SDK_VERSION}`,
-      },
-      ...options,
-    } as InternalClientPipelineOptions);
-
     this.hubName = hubName;
 
     const parsedConnection = parseNotificationHubsConnectionString(connectionString);
-    this.baseUrl = parsedConnection.endpoint;
-    this.sasTokenProvider = createTokenProviderFromConnection(
+    // Node doesn't allow change in protocol but browsers do, so doing a string replace
+    this.baseUrl = parsedConnection.endpoint.replace("sb://", "https://");
+    this.sasTokenCredential = createTokenCredentialFromConnection(
       parsedConnection.sharedAccessKey,
       parsedConnection.sharedAccessKeyName
     );
+
+    this.httpClient = createDefaultHttpClient();
+    this.client = getClient(this.baseUrl, {
+      userAgentOptions: {
+        userAgentPrefix: `azsdk-js-messaging-notificationhubs/${constants.SDK_VERSION}`,
```

Comment:
the format is azsdk-js-\<package_name\> so no "messaging-".
We also need to prefix it with any `options.userAgentOptions.userAgentPrefix` if supplied by the user of this library.

```ts
      const packageDetails = `azsdk-js-notificationhubs/${constants.SDK_VERSION}`;
      const userAgentPrefix = options.userAgentOptions?.userAgentPrefix
        ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
        : `${packageDetails}`;
```

Reply:
Fixed